semgrep-core/parsing/Parse_lua_tree_sitter.ml

```diff
@@ -703,34 +700,26 @@ and map_variable_declarator_expr (env : env) (x : CST.variable_declarator): G.ex
        G.IdQualified (name, G.empty_id_info ())
   )
 
-and map_variable_declarator (env : env) (x : CST.variable_declarator): G.entity =
-  let s = (match x with
-    | `Id tok ->
-        G.EId (identifier env tok, G.empty_id_info ()) (* pattern [a-zA-Z_][a-zA-Z0-9_]* *)
-    | `Self tok ->
-        G.EId (identifier env tok, G.empty_id_info ()) (* pattern [a-zA-Z_][a-zA-Z0-9_]* *)
-    | `Prefix_LBRACK_exp_RBRACK (v1, v2, v3, v4) ->
-        let v1 = map_prefix env v1 in
-        let v2 = token env v2 (* "[" *) in
-        let v3 = map_expression env v3 in
-        let v4 = token env v4 (* "]" *) in
-        let _qual = G.QExpr (v1, v4) in
-        let expr = G.ArrayAccess (v1, (v2, v3, v4)) in
-        G.EDynamic expr
-    | `Field_exp (v1, v2, v3) ->
-        let v1 = map_prefix env v1 in
-        let v2 = token env v2 (* "." *) in
-        let v3 =
-          identifier env v3 (* pattern [a-zA-Z_][a-zA-Z0-9_]* *)
-        in
-        let qual = G.QExpr (v1, v2) in
-        let name = (v3, { G.name_qualifier = Some qual; G.name_typeargs = None }) in
-        G.EName name
-  )
-  in
-  { G.name = s; attrs = []; tparams = [] }
-
-
+and map_variable_declarator (env : env) (x : CST.variable_declarator): G.expr =
+  match x with
+  | `Id tok ->
+      G.Id (identifier env tok, G.empty_id_info ()) (* pattern [a-zA-Z_][a-zA-Z0-9_]* *)
+  | `Self tok ->
```

Comment:
Self is not part of variable_declarator anymore in the latest tree-sitter-lua. I've updated
ocaml-tree-sitter-lang and semgrep in https://github.com/returntocorp/semgrep/pull/2405
you probably need to merge develop to get the latest and resubmit a PR.